src/Messages/OWSMessageSender.m

```diff
@@ -862,8 +865,7 @@ - (void)sendMessage:(TSOutgoingMessage *)message
                         return failureHandler(error, YES);
                     }
 
-                    [self handleMismatchedDevices:serializedResponse recipient:recipient];
-                    retrySend();
+                    [self handleMismatchedDevices:serializedResponse recipient:recipient completion:retrySend];
```

Comment:
retrySend already dispatches async to the send queue. Do you think that concern should be moved?